packages/lab/src/__tests__/__e2e__/overlay/Overlay.cy.tsx

```diff
@@ -3,55 +3,113 @@ import * as overlayStories from "@stories/overlay/overlay.stories";
 import { checkAccessibility } from "../../../../../../cypress/tests/checkAccessibility";
 
 const composedStories = composeStories(overlayStories);
-const { OverlayTop, OverlayRight, OverlayBottom, OverlayLeft } =
-  composedStories;
+const { Default, OverlayRight, OverlayBottom, OverlayLeft } = composedStories;
 
 describe("GIVEN an Overlay", () => {
   checkAccessibility(composedStories);
 
+  describe("WHEN rendered", () => {
+    it("THEN it should show Overlay on trigger element press", () => {
+      cy.mount(<Default />);
+
+      cy.realPress("Tab").realPress("Enter");
+      cy.findByRole("dialog").should("be.visible");
+      // focus goes into floating element
+      cy.findByRole("button", { name: /Close Overlay/i }).should("be.focused");
+    });
+
+    it("THEN it should remain open on repeated trigger element press", () => {
+      cy.mount(<Default />);
+
+      cy.realPress("Tab").realPress("Enter");
+      cy.findByRole("dialog").should("be.visible");
+      cy.findByText(/Show Overlay/i).realClick();
+      cy.findByRole("dialog").should("be.visible");
+    });
+
+    it("THEN it should dismiss on Esc key press", () => {
+      cy.mount(<Default />);
+
+      cy.realPress("Tab").realPress("Enter");
+      cy.findByRole("dialog").should("be.visible");
+      cy.realPress("Escape");
+      cy.findByRole("dialog").should("not.exist");
+      // focus goes back to trigger element on floating element close
+      cy.findByRole("button", { name: /Show Overlay/i }).should("be.focused");
+    });
+
+    it("THEN it should remain open until outside Overlay click or close button click", () => {
+      cy.mount(<Default />);
+
+      cy.findByRole("button", { name: /Show Overlay/i }).realClick();
+      cy.findByRole("dialog").should("be.visible");
+      cy.get(".saltOverlay-closeButton").realClick();
```

Comment:
Let's just use one way of getting to the button for consistency
```suggestion
      cy.findByRole("button", { name: /Show Overlay/i }).realClick();
```